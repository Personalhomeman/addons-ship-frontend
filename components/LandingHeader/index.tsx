import { ReactNode } from 'react';

import css from './style.scss';

interface TitleProps {
  iconUrl: string;
  children: ReactNode;
}

export default ({ iconUrl, children }: TitleProps) => {
  return (
    <div className={css.titleWrapper}>
      <h1 className={css.title}>{children}</h1>
    </div>
  );
};
