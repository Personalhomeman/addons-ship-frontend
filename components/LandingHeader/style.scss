@import '../../assets/style/mixins';
@import '../../assets/style/dimensions';
@import '../../assets/style/colors';

.titleWrapper {
  display: flex;
  align-items: center;
  padding: 4 * $grid-size;
  background-image: url(/static/header-bg.svg);
  background-size: cover;

  .icon {
    margin-right: $grid-size;

    svg {
      fill: white;
    }
  }

  .title {
    min-width: 0;
    color: $white;
    text-overflow: ellipsis;
    word-wrap: break-word;
    // TODO: replace with element from UI library
    line-height: 1.5;
    font-size: 34px;
    letter-spacing: 0.2px;
    font-weight: 700;
    font-style: normal;
    font-stretch: normal;
  }

  @include width-above-screen-small {
    padding-left: 8 * $grid-size;
  }
}
