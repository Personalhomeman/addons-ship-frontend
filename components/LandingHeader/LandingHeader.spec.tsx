import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import LandingHeader from '.';

describe('LandingHeader', () => {
  it('renders correctly', () => {
    const tree = toJSON(shallow(<LandingHeader iconUrl='https://www.bitrise.io/assets/svg/logo-bitrise.svg'>My app</LandingHeader>));
    expect(tree).toMatchSnapshot();
  });
});
