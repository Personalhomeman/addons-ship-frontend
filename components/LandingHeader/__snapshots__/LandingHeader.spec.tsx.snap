// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingHeader renders correctly 1`] = `
<div
  className="titleWrapper"
>
  <InlineSVG
    baseURL=""
    cacheGetRequests={false}
    className="icon"
    onLoad={[Function]}
    src="https://www.bitrise.io/assets/svg/logo-bitrise.svg"
    supportTest={[Function]}
    uniquifyIDs={true}
    wrapper={[Function]}
  />
  <h1
    className="title"
  >
    My app
  </h1>
</div>
`;
